e2e/tests/components/badge/badge.e2e-spec.ts

```diff
@@ -0,0 +1,169 @@
+import { anchorData, buttonData } from '../../../pages/app/badge/badge.testpage.data';
+import { imageCompare } from '../common/image-compare';
+import { BadgePage } from './badge.po.spec';
+
+describe('Badge Tests', () => {
+    let page: BadgePage;
+
+    beforeEach(async () => {
+        page = new BadgePage();
+        await page.getPage();
+    });
+
+    it('should have the correct initial state', async () => {
+        // check the anchor initial state is correct
+        const anchorClasses = await page.anchor.getAttribute('class');
+        const anchorBadgeClasses = await page.anchorBadge.getAttribute('class');
+
+        // check content is correctly displayed
+        expect(await page.anchorBadge.getText()).toBe(anchorData.content);
+
+        // check all classes that are expected are present
+        expect(anchorClasses).toContain('ux-badge-container');
+        expect(anchorClasses).toContain(`ux-badge-${anchorData.size}`);
+        expect(anchorClasses).toContain('ux-badge-overlap');
+        expect(anchorClasses).toContain(`ux-badge-${anchorData.horizontalPosition}`);
+        expect(anchorClasses).toContain(`ux-badge-${anchorData.verticalPosition}`);
+
+        expect(anchorBadgeClasses).toContain('ux-badge');
+
+        // // check text and background colors
+        expect(await page.anchorBadge.getCssValue('color')).toBe('rgba(0, 0, 0, 1)');
+        expect(await page.anchorBadge.getCssValue('background-color')).toBe('rgba(0, 167, 162, 1)');
+
+        // check not hidden
+        expect(await page.anchorBadge.getCssValue('display')).toBe('block');
+
+        // check the button initial state
+        const buttonClasses = await page.button.getAttribute('class');
+        const buttonBadgeClasses = await page.buttonBadge.getAttribute('class');
+
+        // check content is correctly displayed
+        expect(await page.buttonBadge.getText()).toBe(buttonData.content);
+
+        // check all classes that are expected are present
+        expect(buttonClasses).toContain('ux-badge-container');
+        expect(buttonClasses).toContain(`ux-badge-${buttonData.size}`);
+        expect(buttonClasses).toContain('ux-badge-overlap');
+        expect(buttonClasses).toContain(`ux-badge-${buttonData.horizontalPosition}`);
+        expect(buttonClasses).toContain(`ux-badge-${buttonData.verticalPosition}`);
+
+        expect(buttonBadgeClasses).toContain('ux-badge');
+
+        // check text and background colors
+        expect(await page.buttonBadge.getCssValue('color')).toBe('rgba(255, 255, 255, 1)');
+        expect(await page.buttonBadge.getCssValue('background-color')).toBe('rgba(2, 86, 98, 1)');
+
+        // check not hidden
+        expect(await page.buttonBadge.getCssValue('display')).toBe('block');
+
+        expect(await imageCompare('badge-initial')).toEqual(0);
+    });
+
+    it('should be able to create an empty badge', async () => {
```

Comment:
I think most of these tests should have been Karma tests, and as noted above we should leverage the e2e tests mainly to check the visuals of the directive. Protractor tests are notably slower than the Karma tests. Karma tests are also easier to manipulate, so you don't need extra buttons to change states.